src/pydex_app/orderbook.py

```diff
@@ -3,30 +3,133 @@
 
 author: officialcryptomaster@gmail.com
 """
+from zero_ex.dev_utils import abi_utils
+from zero_ex.order_utils import asset_data_utils as adu
+from pydex_app.constants import DEFAULT_ERC20_DECIMALS, DEFAULT_PAGE, DEFAULT_PER_PAGE
+from pydex_app.constants import MAX_INT_STR, SELECTOR_LENGTH, ZERO_STR
 from pydex_app.database import PYDEX_DB as db
 from pydex_app.db_models import SignedOrder
-from utils.miscutils import paginate
+from utils.miscutils import normalize_query_param, paginate, to_api_order
 
 
 class Orderbook:
     """Abstraction for orderbook of signed orders"""
 
+    def __init__(self):
+        pass
+
+    @classmethod
+    def get_order_by_hash_if_exists(cls, order_hash):
+        """Retrieves a specific order by orderHash.
+
+        Keyword arguments:
+        order_hash -- string hash of the signed order to be queried
+        """
+        signed_order_if_exists = SignedOrder.query.get_or_404(normalize_query_param(order_hash))
+        print(signed_order_if_exists.order_status)
+        if signed_order_if_exists.order_status < 3:
+            raise Exception(f"order {order_hash} is INVALID. OrderStatus: {signed_order_if_exists.order_status}")
+        if signed_order_if_exists.order_status > 3:
+            raise Exception(
+                f"order {order_hash} has either been filled, cancelled or has expired.\
+                OrderStatus: {signed_order_if_exists.order_status}"
+            )
+        return to_api_order(signed_order_if_exists.to_json())
+
+    @classmethod
+    def get_asset_pairs(
+        cls,
+        asset_data_a=None,
```

Comment:
Why not stick with notation `maker_asset_data` and `taker_asset_data`?

Reply:
http://sra-spec.s3-website-us-east-1.amazonaws.com/#operation/getAssetPairs